fastapi/dependencies/utils.py

```diff
@@ -171,6 +171,30 @@ def is_scalar_sequence_field(field: Field) -> bool:
     return False
 
 
+def get_typed_signature(call: Callable) -> inspect.Signature:
```

Comment:
I originally wanted to use `typing.get_type_hints` directly, but ultimately this implementation seemed cleaner and follows `typing.get_type_hints` where it matters. It also uses `pydantic.utils` to handle differences between the `typing` api for python 3.6 and 3.7 (for `ForwardRef` and `evaluate_forwardref` anyway), and to deal with the fact that the relevant objects are technically private in `typing`. Since `pydantic` has made them public, it seems reasonable to let `pydantic` worry about ensuring the public API is stable.